<?xml version="1.0"?>
<launch>
  <!-- 使用模拟器进行演示 -->
  <arg name="use_sim" default="true" />
  <arg name="world_name" default="$(find lidar_vision_detection)/worlds/urban.world" />
  <arg name="rviz_config" default="$(find lidar_vision_detection)/rviz/detection_visualization.rviz" />
  
  <!-- 启动Gazebo模拟器 -->
  <group if="$(arg use_sim)">
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
      <arg name="world_name" value="$(arg world_name)" />
      <arg name="paused" value="false" />
      <arg name="use_sim_time" value="true" />
      <arg name="gui" value="true" />
      <arg name="headless" value="false" />
      <arg name="debug" value="false" />
    </include>
  </group>

  <!-- 启动视觉检测节点 -->
  <include file="$(find lidar_vision_detection)/launch/vision_detection.launch">
    <arg name="image_topic" value="/camera/image_raw" />
    <arg name="points_topic" value="/velodyne_points" />
  </include>
  
  <!-- 启动RViz可视化 -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rviz_config)" />

  <!-- 启动示例接收节点，展示如何使用检测结果 -->
  <node pkg="lidar_vision_detection" type="detection_subscriber_example.py" name="detection_subscriber_example" output="screen">
    <param name="min_confidence" value="0.6" />
    <param name="target_classes" value="person,car,truck" />
  </node>
</launch> 